crates/bevy_ecs/src/query/direct.rs

```diff
@@ -0,0 +1,288 @@
+use crate::{
+    component::Component,
+    entity::Entity,
+    query::{
+        Fetch, FilterFetch, QueryEntityError, QueryIter, QueryState, ReadOnlyFetch, WorldQuery,
+    },
+    system::QueryComponentError,
+    world::{Mut, World},
+};
+use bevy_tasks::TaskPool;
+use std::any::TypeId;
+
+// FIXME: this is a copy of [Query] that owns its [QueryState]. They should be merged.
```

Comment:
What are the barriers to merging these?

Reply:
`Query` has a reference to `QueryState` while `DirectQuery` owns its `QueryState`.
Also, `Query` sets the global and system counter to its own in `get_component_unchecked_mut` (but I'm not very pleased with that).

Reply:
`DirectQuery` is removed in 7d38cc4426abe99143cc479bf5befb63de8020c0